src/tally_filter_header.F90

```diff
@@ -0,0 +1,105 @@
+module tally_filter_header
+
+  use constants,       only: MAX_LINE_LEN
+  use particle_header, only: Particle
+
+  use hdf5
+
+  implicit none
+
+!===============================================================================
+! TALLYFILTER describes a filter that limits what events score to a tally. For
+! example, a cell filter indicates that only particles in a specified cell
+! should score to the tally.
+!===============================================================================
+
+  type, abstract :: TallyFilter
+    integer :: n_bins = 0
+  contains
+    procedure(get_next_bin_),  deferred :: get_next_bin
+    procedure(to_statepoint_), deferred :: to_statepoint
+    procedure                           :: to_summary => to_summary_
```

Comment:
To be a little more consistent with other instances of non-deferred type-bound procedures, I'd personally make this `to_summary => filter_to_summary`. Same with initialize below.